iosApp/iosApp/Pages/StopDetails/StopDetailsFilteredDepartureDetails.swift

```diff
@@ -140,9 +143,28 @@ struct StopDetailsFilteredDepartureDetails: View {
         .onChange(of: tripFilter) { tripFilter in
             selectedDepartureFocus = tiles.first { $0.id == tripFilter?.tripId }?.id ?? cardFocusId
         }
+        .onChange(of: patternsByStop) { patternsByStop in
+            patternsHere = patternsHere(patternsByStop)
+        }
+        .onAppear { setAlertSummaries() }
+        .onChange(of: stopDetailsVM.global) { _ in setAlertSummaries() }
+        .onChange(of: alerts) { _ in setAlertSummaries() }
+        .onChange(of: downstreamAlerts) { _ in setAlertSummaries() }
+        .onChange(of: stopId) { _ in setAlertSummaries() }
+        .onChange(of: stopFilter.directionId) { _ in setAlertSummaries() }
+        .onChange(of: patternsHere) { _ in setAlertSummaries() }
+        .onChange(of: now) { _ in setAlertSummaries() }
         .ignoresSafeArea(.all)
     }
 
+    func patternsHere(_ patternsByStop: PatternsByStop) -> [RoutePattern] {
+        // RealtimePatterns.patterns is a List<RoutePattern?> but that gets bridged as [Any] for some reason
```

Comment:
Pulled comment from another place we are doing this. This will go away once we aren't using PatternsByStop anymore.

Reply:
The reason is that generics over optionals aren’t possible in Objective-C and therefore get lost in Swift. I wish frequently that Kotlin had actual Swift interop.